### PR TITLE
Update atomicity guarantees

### PR DESCRIPTION
**Pull request by gonzalobg**
Updates atomicity guarantees for CUDA 12.4, which added support for 128-bit atomic operations at system scope.

### CODE REVIEW

**Comment by gonzalobg** [comment]
@daniellustig for review.

**Comment by miscco** [comment]
Ditto, we want to incorporate those changes into #1753

File: libcudacxx/docs/extended_api/memory_model.md [hunk 1]
```diff
@@ -53,10 +53,10 @@ the scope is `thread_scope_system` and:
 
 - it affects an object in [unified memory] and [`concurrentManagedAccess`] is `1`, **or**
 - it affects an object in CPU memory and [`hostNativeAtomicSupported`] is `1`, **or**
-- it is a load or store that affects a naturally-aligned object of sizes `1`, `2`, `4`, or `8` bytes on [mapped memory], **or**
+- it is a load or store that affects a naturally-aligned object of sizes `1`, `2`, `4`, `8`, or `16` bytes on [mapped memory], **or**
```
Comment (miscco) [inline libcudacxx/docs/extended_api/memory_model.md:56]: We should try and make sure that it is obvious that `16` is only 12.4 onwards

Comment (gonzalobg) [reply libcudacxx/docs/extended_api/memory_model.md:56]: There are no 16-byte wide system scope atomic loads or stores before that version, i.e., this holds retroactively for all CUDA versions, and that includes from the version in which this operations were made available, and onwards.

File: libcudacxx/docs/extended_api/memory_model.md [hunk 2]
```diff
@@ -53,10 +53,10 @@ the scope is `thread_scope_system` and:
...
 - it affects an object in GPU memory and only GPU threads access it.
 
-[mapped memory]: https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#mapped-memory
+> **Note**: If `hostNativeAtomicSupported` is `0`, atomic load or store operations at system scope that affect a naturally-aligned 16-byte wide object in [unified memory] or [mapped memory] require system support. NVIDIA is not aware of any system that lacks this support and there is no CUDA API query available to detect such systems.
```
Comment (daniellustig) [inline libcudacxx/docs/extended_api/memory_model.md:59]: > NVIDIA is not aware of any system that lacks this support and there is no CUDA API query available to detect such systems.

NVIDIA is not aware of any system that lacks this support, but there is no CUDA API query available to detect such systems.

Comment (gonzalobg) [reply libcudacxx/docs/extended_api/memory_model.md:59]:
```suggestion
> **Note**: If `hostNativeAtomicSupported` is `0`, atomic load or store operations at system scope that affect a naturally-aligned 16-byte wide object in [unified memory] or [mapped memory] require system support. NVIDIA is not aware of any system that lacks this support, but there is no CUDA API query available to detect such systems.
```

**Pull request closed without merging**